tests/functional/spec/dates.spec.js

```diff
@@ -239,4 +240,73 @@ describe('Date checks', function() {
             // Then the day subfield should gain the focus
             expect(browser.hasFocus(DatesPage.dayInput.selector)).to.be.true
       })
+
+      it('Given the test_dates survey is selected when dates are entered then the summary screen shows the conditional piped dates entered formatted', function() {
+
+            // Given the test_dates survey is selected
+            openQuestionnaire('test_conditional_dates.json')
+
+            // When dates are entered
+            DatesPage.setFromReportingPeriodDay(1)
+              .setFromReportingPeriodMonth(3)
+              .setFromReportingPeriodYear(2016)
+              .setToReportingPeriodDay(3)
+              .setToReportingPeriodMonth(5)
+              .setToReportingPeriodYear(2017)
+              .setMonthYearMonth(4)
+              .setMonthYearYear(2018)
+              .setDateOfBirthDay(4)
+              .setDateOfBirthMonth(1)
+              .setDateOfBirthYear(1999)
+              .setStartReportingPeriodDay(11)
+              .setStartReportingPeriodMonth(10)
+              .setStartReportingPeriodYear(2017)
+              .setEndReportingPeriodDay(3)
+              .setEndReportingPeriodMonth(12)
+              .setEndReportingPeriodYear(2017)
+              .submit()
+              .setDatesAnswer(1)
+              .submit()
```

Comment:
If a page has been added to the survey, then a page object should be generated and this submit should be called on that.